src/codec.rs

```diff
@@ -1058,6 +1058,18 @@ impl Encode for str {
 	}
 }
 
+impl Decode for Box<str> {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+		Ok(String::decode(input)?.into_boxed_str())
+	}
+}
+
+impl<T: Decode> Decode for Box<[T]> {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+		Ok(Vec::decode(input)?.into_boxed_slice())
```

Comment:
So this is correct since `&[T]` implements `EncodeLike<Vec<T>>`?

Reply:
Every array is encoded the same way in Scale. First the length as compact and then the data.

Reply:
Will `Vec<T>::decode` create vector where `capacity` == `len`? 
If not, I will make an impl comparable to `Box<T>`

Otherwise, there is no downside in decoding to `Vec<T>` and then converting to `Box<[T]>`

Reply:
> Will `Vec<T>::decode` create vector where `capacity` == `len`?

Not always; only if it takes the optimized code path. (So e.g. for `u8` it should, but not for any arbitrary `T`.)

Reply:
fixed, please check again